src/shared/database/index.ts

```diff
@@ -79,6 +100,159 @@ class Pindexer {
+        q: row.q,
+        rowid: row.rowid
+      },
+      opening_reserves: {
+        height: row.height,
+        position_id: row.position_id,
+        reserves_1: row.reserves_1,
+        reserves_2: row.reserves_2,
+        rowid: row.opening_reserves_rowid,
+        time: row.time,
+      },
+      latest_reserves: latestReserves,
+    };
+  }
+
+  async getPositionExecutionsWithReserves(positionId: Buffer): Promise<{ items: ExecutionWithReserves[], skippedRows: number }> {
+    const maxCount = 100;
+    
+    // First, get the total count
+    const totalCount = await this.db
+      .selectFrom('dex_ex_position_executions')
+      .select(sql`count(*)`.as('count'))
+      .where('position_id', '=', positionId)
+      .executeTakeFirstOrThrow();
+
+    // Then get the actual items
+    const results = await this.db
+      .selectFrom('dex_ex_position_executions as executions')
+      .innerJoin(
+        'dex_ex_position_reserves as reserves',
+        'executions.reserves_rowid',
+        'reserves.rowid'
+      )
+      .selectAll('executions')
+      .selectAll('reserves')
+      .where('executions.position_id', '=', positionId)
+      .orderBy('executions.height', 'desc')
+      .orderBy('executions.rowid', 'desc')
+      .limit(maxCount)
+      .execute();
+
+    const items = results.map(row => ({
+      execution: {
+        context_asset_end: row.context_asset_end,
+        context_asset_start: row.context_asset_start,
+        delta_1: row.delta_1,
+        delta_2: row.delta_2,
+        fee_1: row.fee_1,
+        fee_2: row.fee_2,
+        height: row.height,
+        lambda_1: row.lambda_1,
+        lambda_2: row.lambda_2,
+        position_id: row.position_id,
+        reserves_rowid: row.reserves_rowid,
+        rowid: row.rowid,
+        time: row.time,
+      },
+      reserves: {
+        height: row.height,
+        position_id: row.position_id,
+        reserves_1: row.reserves_1,
+        reserves_2: row.reserves_2,
+        rowid: row.reserves_rowid,
+        time: row.time,
+      }
+    }));
+
+    return {
+      items,
+      skippedRows: Math.max(0, Number(totalCount.count) - maxCount)
+    };
+  }
+
+  async getPositionWithdrawals(positionId: Buffer): Promise<Selectable<DexExPositionWithdrawals>[]> {
+    return this.db
+      .selectFrom('dex_ex_position_withdrawals')
+      .selectAll()
+      .where('position_id', '=', positionId)
+      .orderBy('height', 'desc')
+      .orderBy('rowid', 'desc')
+      .execute();
+  }
+
+  async getPositionVolumeAndFees(positionId: Buffer): Promise<VolumeAndFees[]> {
+    const results = await this.db
+      .selectFrom('dex_ex_position_executions')
+      .select([
+        'context_asset_start',
+        'context_asset_end',
+        sql`sum(delta_1 + lambda_1)`.as('volume1'),
+        sql`sum(delta_2 + lambda_2)`.as('volume2'),
+        sql`sum(fee_1)`.as('fees1'),
+        sql`sum(fee_2)`.as('fees2'),
+        sql`CAST(count(*) AS INTEGER)`.as('executionCount'),
```

Comment:
suggestion: this would be better typed with the help of generics:

```ts
sql<number>`sum(delta_1 + lambda_1)`.as('volume1'),
```

meaning the returned `volume1` field would be of type `number`